error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StreamRules.java

```diff
@@ -167,6 +167,34 @@ Stream<R> after(Stream<T> stream, Function<? super S, ? extends Stream<? extends
     }
   }
 
+  /** Apply filtering before sorting to reduce the number of elements to sort. */
+  static final class SortAfterFilter<T> {
```

Comment:
(We still have to automate this, but) our current naming scheme would dictate that this class is named `StreamFilterSorted`.